src/style/style.js

```diff
@@ -365,9 +365,7 @@ class Style extends Evented {
         } else {
             delete this.stylesheet.projection;
         }
-        if (!this.map._explicitProjection) {
-            this.map._updateProjection();
-        }
+        this._updateMapProjection();
```

Comment:
This change may impact performance as it triggers an extra `setProjection` worker event in cases where `map.explicitProjection !== null` (i.e. when the map remains unchanged)

Reply:
Added conditional back in